AsyncDisplayKit/ASMultiplexImageNode.mm

```diff
@@ -456,8 +455,8 @@ - (void)_loadNextImage
     }];
   }
   // Likewise, if it's a iOS 8 Photo asset, we need to fetch it accordingly.
-  else if (AS_AT_LEAST_IOS8 && [[nextImageURL scheme] isEqualToString:kPHAssetURLScheme]) {
-    [self _loadPHAssetWithIdentifier:nextImageIdentifier URL:nextImageURL completion:^(UIImage *image, NSError *error) {
+  else if (ASPhotosFrameworkImageRequest *request = nextImageURL.asyncdisplaykit_photosRequest) {
```

Comment:
That's not quite the flow — the category on NSURL is just a convenience wrapper for ASPhotosFrameworkRequest.requestWithURL which does the ph:// parsing like you mentioned. I'd be fine with ditching it since it only saves you like one line tops. But there's no magic here! Except the sublime magic of creation.

And yes! There is at least one use case for remote images intermingled with local images – I'm working on an app that lets you create and configure photo-based content offline, and then it is synced in the background. So we'll have a flow of photos where some are local and some are remote and that might even change while the image is onscreen. 

Sent from my iPhone

> On Sep 27, 2015, at 12:01 PM, appleguy <notifications@github.com> wrote:
> 
> In AsyncDisplayKit/ASMultiplexImageNode.mm:
> 
> > @@ -456,8 +455,8 @@ - (void)_loadNextImage
> >      }];
> >    }
> >    // Likewise, if it's a iOS 8 Photo asset, we need to fetch it accordingly.
> > -  else if (AS_AT_LEAST_IOS8 && [[nextImageURL scheme] isEqualToString:kPHAssetURLScheme]) {
> > -    [self _loadPHAssetWithIdentifier:nextImageIdentifier URL:nextImageURL completion:^(UIImage *image, NSError *error) {
> > +  else if (ASPhotosFrameworkImageRequest *request = nextImageURL.asyncdisplaykit_photosRequest) {
> I realize that this pattern (.asyncdisplaykit_) is used in a few places, but I think those usages are usually quite necessary (like extensions to CALayer) without a clean way to avoid it.
> 
> In this case, is it not correct to say the flow is:
> 1. User creates a PhotosRequest object
> 2. User manually requests a URL from that object, and provides it to ASMultiplexImageNode
> 3. User discards reference to PhotosRequest
> 4. Internals of ASMultiplexImageNode revive the reference by grabbing associated object through magically-extended NSURL object
> 
> This is the only part of the diff that doesn't feel right to me. Even parsing ph:// seems a bit better :). Are there any real-world cases where Photos framework images would be mixed with regular remote URLs? If not, what do you think about providing a load method that takes the ASPhotosFrameworkImageRequest directly?
> 
> —
> Reply to this email directly or view it on GitHub.
>